samples/snippets/src/test/java/com/beta/automl/CancelOperationTest.java

```
@@ -0,0 +1,89 @@
+import java.io.IOException;
+import java.io.PrintStream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class CancelOperationTest {
+
+  private static final String PROJECT_ID = System.getenv("AUTOML_PROJECT_ID");
+
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+  private PrintStream originalPrintStream;
+
+  private static String requireEnvVar(String varName) {
+    String value = System.getenv(varName);
+    assertNotNull(
+        "Environment variable " + varName + " is required to perform these tests.",
+        System.getenv(varName));
+    return value;
+  }
+
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    requireEnvVar("AUTOML_PROJECT_ID");
+  }
+
+  @Before
+  public void setUp() {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    originalPrintStream = System.out;
+    System.setOut(out);
+  }
+
+  @After
+  public void tearDown() {
+    // restores print statements in the original method
+    System.out.flush();
+    System.setOut(originalPrintStream);
+  }
+
+  @Test
+  public void testCancelOperation() throws IOException {
+    String operationFullPathId =
+        String.format(
+            "projects/%s/locations/%s/operations/%s", PROJECT_ID, "us-central1",
+            "TCN0000000000");
+    // Any cancelled operation on models or datasets will be hidden once the operations are flagged
+    // as failed operations
+    // which makes them hard to delete in the teardown.
+    try {
+      CancelOperation.cancelOperation(operationFullPathId);
+      String got = bout.toString();
+      assertThat(got).contains("not found");
+    } catch (NotFoundException | StatusRuntimeException | InterruptedException e) {
+      assertThat(e.getMessage()).contains("not found");
+    }
```

Ok, but really not required or helpful as the framework should catch these.

---

Hmm, I do not understand what you mean by that. Could you elaborate on what should be done?

---

If I start w/ OK, i'm only observing something, not requiring a change, it's ok if you want to change it, but I'll approve (or may have approved) w/o it.

That said, Junit4, will catch and report all exceptions that happen.  Your code is eating the exceptions, and then gating on "not found" being present.  (ie. it's ok if we get an exception as long as "not found" is in the output, so I guess there wasn't really a need for me to comment - it just seems a bit weird.  (which I'm ok with in tests)